noir_stdlib/src/hash/poseidon2.nr

```diff
@@ -100,7 +100,7 @@ impl Poseidon2 {
 
     fn hash_internal<N>(input:[Field;N], in_len:u32, is_variable_length: bool) -> Field
     {
-        let iv : Field = (in_len as Field)*18446744073709551616;
+        let iv : Field = (in_len as Field)*18446744073709551616;    // iv = (in_len << 64)
```

Comment:
```suggestion
        let two_pow_64 = 18446744073709551616;
        let iv : Field = (in_len as Field) * two_pow_64;
```

I think that it's best to document this by using a named variable instead of a magic number here.

Reply:
done